client/state/posts/counts/selectors.js

```diff
@@ -21,7 +21,7 @@ export function isRequestingPostCounts( state, siteId, postType ) {
  * @param  {object} state    Global state tree
  * @param  {number} siteId   Site ID
  * @param  {string} postType Post type
- * @returns {object}          Post counts, keyed by status
+ * @returns {Object.<string, number>}          Post counts, keyed by status
```

Comment:
Not related to this PR but may be we can use `@typescript-eslint/parser` as the default parser for ESLint. Then we could use TS types for such types.
So, instead of `Object.<string, number>`, we could use `Record<string, number>`.

Reply:
That would be wonderful. 💯

Reply:
I have created a PR for it #64985